developer_guides/topology2/topology2.rst

```diff
@@ -0,0 +1,895 @@
+
+.. code-block:: bash
+
+	Object.Widget.pga."1" {
+		index 5
+
+		# volume control'
+		Object.Control.mixer."1" {
+				name "Master Control Volume 5"
+			}
+		}
+	}
+
+Now the mixer control object is assigned the name "Master Control Volume 5" indicating that it belongs to
+the pga widget in the pipeline with index 5.
+
+
+1.2.5 Nested Objects
+********************
+Objects can also be instantiated as child objects within other object instances. For example, a
+switch control can be added to pga widget objects during instantiation as follows:
+
+.. code-block:: bash
+
+	Object.Widget.pga."1" {
+		index 5
+		
+		# volume control
+		Object.Control.mixer."1" {
+				name "Master Control Volume 5"
+			}
+		}
+
+		# mute control
+		Object.Control {
+			mixer."2" {
+				name "Mute Switch Control"
+				max 1
+			}
+		}
+	}
+
+Note how the "unique" attribute for the 2 mixer control objects differ to keep the mixer instances unique.
+
+1.2.6 Recursive object attribute inheritance
+********************************************
+Objects can be nested within objects that are nested within other objects themselves. In this case, the attribute
+values cam be inherited all the way from the top-level parent object. For example, consider the following class
+definition for volume-playback pipeline:
+
+.. code-block:: bash
+
+	Class.Pipeline."volume-playback" {
+		# Other attributes and qualifiers ommitted for simplicity
+		DefineAttribute."index" {}
+
+		DefineAttribute."format" {
+			type	"string"
+		}
+
+		# pipeline objects
+		Object.Widget {
+			# Other objects ommitted for simplicity
+
+			pga."1" {}
+		}
+	}
+
+Note that the pga widget object above has no index attribute value. An object of volume-playback
+class would be instantiated as:
+
+.. code-block:: bash
+
+	Object.Pipeline.volume-playback.1 {
+		index 1
+		format s24le
+	}
+
+This would ensure that all child objects within the volume-playback object will inherit the
+index attribute value from it. So the pga widget object will have the same index and by the same
+rule, the mixer control object within the pga widget object would also have the same index attribute
+value of "1".
+
+1.2.7 Setting child object attributes deep down in the parent object tree
+*************************************************************************
+In section 1.2.2, we saw that we can set child attribute values from its parent object instance.
+For example, the mixer control object's name can be set from the pga widget object instace. This
+can be applied further and it is possible to set the mixer control name from the parent object of
+the pga object. Consider the volume playback object instance in the previous section. We can set the
+mixer control name for the pga object as follows:
+
+.. code-block:: bash
+
+	Object.Pipeline.volume-playback.1 {
+		index 1
+		format s24le
+		Object.Widget.pga.1 {
+			Object.Control.mixer.1 {
```

Comment:
what do the ".1" mean or represented in the hierarchy?

maybe we should have an example with multiple objects in the same pipe to show how this works